pulsar-broker/src/main/java/org/apache/pulsar/compaction/TwoPhaseCompactor.java

```diff
@@ -145,14 +149,17 @@ private void phaseOneLoop(RawReader reader,
                     Pair<String, Integer> keyAndSize = extractKeyAndSize(m);
                     if (keyAndSize != null) {
                         if (keyAndSize.getRight() > 0) {
-                            latestForKey.put(keyAndSize.getLeft(), id);
+                            MessageId old = latestForKey.put(keyAndSize.getLeft(), id);
+                            replaceMessage = old != null;
                         } else {
                             deletedMessage = true;
                             latestForKey.remove(keyAndSize.getLeft());
                         }
                     }
                 }
-
+                if (replaceMessage || deletedMessage) {
+                    mxBean.addCompactRemovedEvent(reader.getTopic());
+                }
```

Comment:
Maybe the `replaceMessage` could be re-modified. For example, in a batch loop, at first, it could be true, when meeting a new key it will change to false.

I'm not sure about the meaning of the `compactRemoveEvent`, this event indicates one batch compact event or message compact events.

Reply:
`compactRemoveEvent` indicates how many duplicate messages have been removed in this compact task.

Reply:
> Maybe the `replaceMessage` could be re-modified. For example, in a batch loop, at first, it could be true, when meeting a new key it will change to false.
Yes, really, Thanks for the kindly review.